docs/guides/materialize_source_table.md

```diff
@@ -0,0 +1,245 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+# template file: templates/guides/materialize_source_table.md.tmpl
+page_title: "Source Table Migration Guide"
+subcategory: ""
+description: |-
+  Guide for migrating to the new materialize_source_table_{source_type} resources.
+---
+
+# Source versioning: migrating to `materialize_source_table_{source_type}` Resource
+
+In previous versions of the Materialize Terraform provider, source tables were defined within the source resource itself and were considered subsources of the source rather than separate entities.
+
+This guide will walk you through the process of migrating your existing source table definitions to the new `materialize_source_table_{source_type}` resource.
```

Comment:
I'm confused on whether this guide is meant to explain how to migrate using just the manual process (which is outlined below and looks correct), or also to explain how to reconcile your terraform configuration with the results of the 'automatic' migration process that we would do for a customer with a catalog migration. Specifically, the paragraph below sounds like it is explaining what to do to handle the automatic migration process, but it doesn't explain that the manual process would then be unnecessary.

It might be simpler if this guide just explained the manual process and we had a separate one to explain the 'automatic' process. Or we could just assume any terraform user would only do the manual process. @morsapaes thoughts?

Reply:
Good point! I've removed some of the confusing text so that this should focus on the manual migration!

Happy to work on a follow up migration guide for the automated migration if we decide to do that!